### ISSUE TITLE
Forgot Password . Account Recovery option

### ISSUE THREAD

**Issue by hsitaa** [sdelements/lets-chat#351]
if the user forgets his password then its lost no recovery because no admin. So its all previous chats lost. and

    it won't allow creation of new logins with same email id.
    No way to recover password or get the email id free to create new account.

Pretty lock down scenario. In case if want to use in live scenario.
A big road block. coz password recovery is a most common issue at organizational Level.
Features required :
*Password recovery

    Admin account or at least a way to use the email id's whose password has been forgotten.

Is there any way to deal with the issues currently ?

**Comment by bessarabov** [comment]
> A big road block. coz password recovery is a most common issue at organizational Level.

I think that at this level you should not use Local authentication, but instead use  Kerberos / LDAP. In this case the passwords will be stored an managed outside lets-chat.

Concerning your problem. I'm sure that is possible to connect to mongo manually and fix password direct in the database. I don't know the details but I think that is should be pretty easy.

**Comment by sibartlett** [comment]
We hope to have a password reset feature in the next release.

This hasn't been a high priority for ourselves as we use Kerberos and LDAP, but we understand how important it is for others.

If someone wants to contribute to this feature - that'd be even more awesome!

**Comment by funkaoshi** [comment]
Yeah, we fix "local" accounts by fixing things in Mongo directly. It's no ideal, but as noted we manage all our users externally currently.

Also, might just want to merge this with #9.

**Comment by bnvk** [comment]
I'm going to take a stab at implementing this. Wish me luck. Any pointers are welcome ;)

**Comment by funkaoshi** [comment]
That was the royal "we". @hhaidar can point you in the right direction.

**Comment by Fibrewire** [comment]
Could you be more specific on "Yeah, we fix "local" accounts by fixing things in Mongo directly.", im struggling to find/connect/edit the mongoDB

**Comment by treeves** [comment]
Any tips on how best to fix locked-out accounts would be great - or at least a way to toggle account lockout off, so that users can't lock themselves out.

**Comment by hhaidar** [comment]
@treeves you can disable it by adding this to your settings:

```
auth:
  throttling:
    enable: true
```

**Comment by hhaidar** [comment]
@Fibrewire http://www.hacksparrow.com/the-mongodb-tutorial.html

Hopefully we'll release the command line helper tool soon. That should help with this kind of thing.

**Comment by treeves** [comment]
@hhaidar Much appreciated.  That auth throttling flag did it, and looks like that disabled all of the lockout code, as my locked-out user can now get in.

**Comment by geoffwhittington** [comment]
@hhaidar Do you mean to set `enable: false` above?

**Comment by hhaidar** [comment]
@geoffwhittington haha yea, butter brain today ;P

@treeves good stuff! You can also look in defaults.yml if you want to look at loosening up the restrictions

**Comment by treeves** [comment]
(curiously, enable: true worked to disable it.  The inverse inference security model shall fool ALL the hackers!)

**Comment by hhaidar** [comment]
@treeves you definitely want `false` there. It might be setting `threshold` as blank or something. We should fix that :S

**Comment by mabahj** [comment]
Was looking for a guide on how to remove users that are locked out or have forgotten their password. So for future searchers, this is how you remove the user:
```
% ./mongo your_server_name/letschat                                                                                                                                                       [teamchat/mongodb/bin|9:56:24|225s]
MongoDB shell version: 3.0.6
connecting to: your_server_name/letschat
> use letschat
[...]
> show collections
[...]
> db.users.find()
[...]
> db.users.find({username:"the_user"})
[...]
> db.users.remove({username:"the_user"})
[...]
```

**Comment by chetan** [comment]
Adding to the previous comment, a quick and easy way to reset a password instead of nuking the user completely:

```
# Find user as above then generate a new hash
$ node -e "var bcrypt = require('bcryptjs'); bcrypt.hash(process.argv[1], 10, function(err, hash) { console.log('your new password hash is:', hash) });" NEW_PASSWORD

# copy hash and update in db
> db.users.update({username: "the_user", {$set: {password: "$2a$10$abcdefghijklmnopqrstuvwxyz"}})
```
